src/contracts/nft-collection.ts

```diff
@@ -98,11 +103,7 @@ export class NFTCollection extends Erc721<TokenERC721> {
    * const mintedId = tx.id; // the id of the NFT minted
    * ```
    */
-  public signature: Erc721SignatureMinting;
-  /**
-   * @internal
-   */
-  public interceptor: ContractInterceptor<TokenERC721>;
+  public signature = this.signatureMint;
```

Comment:
make sure you cast this like i do for `_mint` below, we know this will always be defined so we can force it to always be defined. 

We don't want ppl to have to do `nftCollection.signature?.mint()`